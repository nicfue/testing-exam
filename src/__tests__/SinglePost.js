import React from 'react';
import { mount, shallow, render } from 'enzyme';
import SinglePost from '../components/SinglePost';

describe('Unit test on SinglePost', ()=>{
  const post = jest.fn();
  const wrapper1 = mount(<SinglePost
    title="heading"
    content="text"
    id="testId"
    author="Zac"
    currentPersona="Zac"
    date="0000000"
    onClick={post}
    />)

    const wrapper2 = mount(<SinglePost
      title="heading"
      content="text"
      id="testId"
      author="Zac"
      currentPersona="Morgana"
      date="0000000"
      onClick={post}
      />)

it('if author and currentPersona is the same, then the button should render', () => {
  expect(wrapper1.find('Button').length).toEqual(1);
})

it('if author and currentPersona are not the same, then the button should not render', () => {
  expect(wrapper2.find('Button').length).toEqual(0);
 })

 it('snapshots when match', ()=> {
   expect(wrapper1).toMatchSnapshot();
 })

 it('snapshots when mismatch', ()=> {
   expect(wrapper2).toMatchSnapshot();
 })

 it('posts should contain a title and content', () => {
   expect(wrapper1.find('article h2').text()).toEqual('heading')
 })

 it('test onclick with id', ()=>{
   wrapper1.find('button').simulate('click');
   expect(post).toBeCalledWith('testId');
 });

})
