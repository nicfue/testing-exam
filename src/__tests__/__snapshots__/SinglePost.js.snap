// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test on SinglePost snapshots when match 1`] = `
<SinglePost
  author="Zac"
  content="text"
  currentPersona="Zac"
  date="0000000"
  id="testId"
  onClick={[Function]}
  title="heading"
>
  <div
    className="w-full shadow p-6 m-6 border rounded relative"
  >
    <article
      className="mb-2"
    >
      <Button
        className="absolute pin-t pin-r mx-4 my-6"
        danger={true}
        onClick={[Function]}
      >
        <button
          className="bg-red-dark hover:bg-red-darker text-white font-bold py-2 px-4 rounded-full absolute pin-t pin-r mx-4 my-6"
          data-test="button"
          onClick={[Function]}
        >
          x
        </button>
      </Button>
      <h2
        className="mb-4 w-4/5"
      >
        heading
      </h2>
      <ReactMarkdown
        className="mb-4 text-grey-darkest markdown"
        escapeHtml={true}
        renderers={Object {}}
        skipHtml={false}
        source="text"
        transformLinkUri={[Function]}
      >
        <div
          className="mb-4 text-grey-darkest markdown"
          key="root-1-1"
        >
          <p
            key="paragraph-1-1"
          >
            text
          </p>
        </div>
      </ReactMarkdown>
      <p
        className="text-grey-dark mb-4"
      >
        Posted by: 
        Zac
         @ 
        0000000
      </p>
    </article>
    <Comments
      currentPersona="Zac"
      postId="testId"
    >
      <div
        className="py-2"
      >
        <h2
          className="text-indigo-darker border-b mb-4"
        >
          Comments
        </h2>
        <CreateNewComment
          author="Zac"
          postId="testId"
          updateComments={[Function]}
        >
          <form
            className="container mx-auto flex flex-col p-6"
            onSubmit={[Function]}
          >
            <label
              className="mb-2"
              htmlFor="comment"
            >
              Comment
            </label>
            <textarea
              className="shadow rounded border mb-2 p-2"
              id="comment"
              name="comment"
              onChange={[Function]}
              value=""
            />
            <input
              className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
              type="submit"
              value="Comment"
            />
          </form>
        </CreateNewComment>
      </div>
    </Comments>
  </div>
</SinglePost>
`;

exports[`Unit test on SinglePost snapshots when mismatch 1`] = `
<SinglePost
  author="Zac"
  content="text"
  currentPersona="Morgana"
  date="0000000"
  id="testId"
  onClick={[Function]}
  title="heading"
>
  <div
    className="w-full shadow p-6 m-6 border rounded relative"
  >
    <article
      className="mb-2"
    >
      <h2
        className="mb-4 w-4/5"
      >
        heading
      </h2>
      <ReactMarkdown
        className="mb-4 text-grey-darkest markdown"
        escapeHtml={true}
        renderers={Object {}}
        skipHtml={false}
        source="text"
        transformLinkUri={[Function]}
      >
        <div
          className="mb-4 text-grey-darkest markdown"
          key="root-1-1"
        >
          <p
            key="paragraph-1-1"
          >
            text
          </p>
        </div>
      </ReactMarkdown>
      <p
        className="text-grey-dark mb-4"
      >
        Posted by: 
        Zac
         @ 
        0000000
      </p>
    </article>
    <Comments
      currentPersona="Morgana"
      postId="testId"
    >
      <div
        className="py-2"
      >
        <h2
          className="text-indigo-darker border-b mb-4"
        >
          Comments
        </h2>
        <CreateNewComment
          author="Morgana"
          postId="testId"
          updateComments={[Function]}
        >
          <form
            className="container mx-auto flex flex-col p-6"
            onSubmit={[Function]}
          >
            <label
              className="mb-2"
              htmlFor="comment"
            >
              Comment
            </label>
            <textarea
              className="shadow rounded border mb-2 p-2"
              id="comment"
              name="comment"
              onChange={[Function]}
              value=""
            />
            <input
              className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
              type="submit"
              value="Comment"
            />
          </form>
        </CreateNewComment>
      </div>
    </Comments>
  </div>
</SinglePost>
`;
